compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp

```diff
@@ -29,28 +31,50 @@ namespace {
 class PolynomialApproximationPass final
     : public impl::PolynomialApproximationPassBase<
           PolynomialApproximationPass> {
+public:
+  using Base::Base;
+
   void runOnOperation() override {
+    using PatternFunction = llvm::function_ref<void(RewritePatternSet &)>;
+    llvm::StringMap<PatternFunction> patternMap = {
+        {"tan", populateExpandTanPattern},
+        {"sinh", populateExpandSinhPattern},
+        {"cosh", populateExpandCoshPattern},
+        {"asinh", populateExpandAsinhPattern},
+        {"acosh", populateExpandAcoshPattern},
+        {"atanh", populateExpandAtanhPattern},
+        {"powf", populateExpandPowFPattern},
+        {"fpowi", populateExpandFPowIPattern},
+        {"erf",
+         [&](RewritePatternSet &mathPatterns) {
+           this->populateErfPattern(mathPatterns);
+         }},
```

Comment:
I think this needs to be a local variable to avoid references to temporaries.

Reply:
the name is indeed overlapped. I've updated it to make a difference.